src/coreclr/jit/importer.cpp

```diff
@@ -4782,6 +4788,390 @@ GenTree* Compiler::impIntrinsic(GenTree*                newobjThis,
     return retNode;
 }
 
+GenTree* Compiler::impSRCSUnsafeIntrinsic(NamedIntrinsic        intrinsic,
+                                          CORINFO_CLASS_HANDLE  clsHnd,
+                                          CORINFO_METHOD_HANDLE method,
+                                          CORINFO_SIG_INFO*     sig)
+{
+    assert(sig->sigInst.classInstCount == 0);
+
+    switch (intrinsic)
+    {
+        case NI_SRCS_UNSAFE_Add:
+        {
+            assert(sig->sigInst.methInstCount == 1);
+
+            // ldarg.0
+            // ldarg.1
+            // sizeof !!T
+            // conv.i
+            // mul
+            // add
+            // ret
+
+            GenTree* op2 = impPopStack().val;
+            GenTree* op1 = impPopStack().val;
+            impBashVarAddrsToI(op1, op2);
+
+            GenTree* size = gtNewIconNode(info.compCompHnd->getClassSize(sig->sigInst.methInst[0]), TYP_INT);
+            impBashVarAddrsToI(size);
+            size = gtNewCastNode(genActualType(TYP_I_IMPL), size, /* uns */ false, TYP_I_IMPL);
+
+            var_types type = impGetByRefResultType(GT_MUL, /* uns */ false, &op2, &size);
```

Comment:
Nit: `size == 1` can skip the multiplication

Reply:
Added a path to skip the multiply if `classSize == 1`